icechunk/src/repository.rs

```diff
@@ -139,8 +139,7 @@ pub struct Repository {
     asset_manager: Arc<AssetManager>,
     virtual_resolver: Arc<VirtualChunkResolver>,
     virtual_chunk_credentials: HashMap<ContainerName, Credentials>,
-    #[serde(skip)]
-    default_commit_metadata: Arc<RwLock<Option<SnapshotProperties>>>,
+    default_commit_metadata: Arc<Mutex<Option<SnapshotProperties>>>,
```

Comment:
Why Mutex over RwLock here?

Reply:
Yeah exactly. And here we can't use the async versions because of serialization. I can use rwlock if @paraseba would prefer tho